### PR TITLE
RMLNQFETCH-19 Support for .net standard

### PR DESCRIPTION
**Pull request by ngbrown**
Pull request for [RMLNQFETCH-19](https://www.re-motion.org/jira/projects/RMLNQFETCH/issues/RMLNQFETCH-19).

### CODE REVIEW

**Comment by MichaelKetting** [comment]
Looks good :) Thanks for the work!

I did find a couple of bits that weren't obivous to anyone not deep into this particular setup and pushed them into https://github.com/re-motion/Relinq-EagerFetching/commits/feature/RMLNQFETCH-19-Support-For-NetStandard. If you want, feel free to go head and rewrite the PR to integrate my updates to your commits :)

I've also uploaded a nightly build of the .NET Standard enabled package: https://www.re-motion.org/jira/secure/attachment/12900/Remotion.Linq.EagerFetching.2.1.0-x-001-25.nupkg
It would be great if you could give it a whirl to make sure everything works with .NET Standard. (I don't have a test setup for that ATM.) I'll do the release after that one.

**Review by MichaelKetting — APPROVED** [review]
Looks good :) Thanks for the work!

I did find a couple of bits that weren't obivous to anyone not deep into this particular setup and pushed them into https://github.com/re-motion/Relinq-EagerFetching/commits/feature/RMLNQFETCH-19-Support-For-NetStandard. If you want, feel free to go head and rewrite the PR to integrate my updates to your commits :)

I've also uploaded a nightly build of the .NET Standard enabled package: https://www.re-motion.org/jira/secure/attachment/12900/Remotion.Linq.EagerFetching.2.1.0-x-001-25.nupkg
It would be great if you could give it a whirl to make sure everything works with .NET Standard. (I don't have a test setup for that ATM.) I'll do the release after that one.

**Comment by ngbrown** [comment]
@MichaelKetting I did re-write your improvements into my pull request.  I've also used your attachment in a NHibernate pull request (nhibernate/nhibernate-core#633) and it seems to be working well enough.  Any chance you could publish the NuGet package on a feed of some sort, even if only as an alpha or beta?

**Comment by MichaelKetting** [comment]
@ngbrown Thanks. I've published a beta-release on nuget.org.

**Comment by hazzik** [comment]
@MichaelKetting can this be released as a release version?

**Comment by MichaelKetting** [comment]
Yes, hopefully. Mainly, I was waiting for feedback on the beta, but now I found an issue last week, too. (https://www.re-motion.org/jira/browse/RMLNQFETCH-20)

**Comment by hazzik** [comment]
Actually, the same happens with ReLinq itself

**Comment by MichaelKetting** [comment]
Yes, it does. The guys at PostSharp found that one (https://www.re-motion.org/jira/browse/RMLNQ-113)

File: UnitTests.Net_4_5/packages.config [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="NUnit" version="2.6.3" targetFramework="net40" />
+  <package id="Remotion.Development.UnitTesting.PrivateInvoke.Sources" version="1.15.23.0" targetFramework="net45" developmentDependency="true" />
+  <package id="Remotion.JetBrains.Annotations.Sources" version="1.15.23.0" targetFramework="net40" developmentDependency="true" />
+  <package id="Remotion.Linq" version="2.0.0" targetFramework="net40" />
```
Comment (hazzik) [inline UnitTests.Net_4_5/packages.config]: This shall be 2.1.1

File: UnitTests.Net_4_5/packages.config [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  <package id="Remotion.Linq.Development" version="2.0.0" targetFramework="net40" />
```
Comment (hazzik) [inline UnitTests.Net_4_5/packages.config]: Same as above

File: Core.Net_4_5/packages.config [hunk 3]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="Remotion.JetBrains.Annotations.Sources" version="1.15.23.0" targetFramework="portable-net45+win+wpa81+wp80" developmentDependency="true" />
+  <package id="Remotion.Linq" version="2.0.0" targetFramework="net40-Client" />
```
Comment (hazzik) [inline Core.Net_4_5/packages.config]: This shall be 2.1.1

**Comment by MichaelKetting** [comment]
@hazzik 
Okay, v2.1.0-beta.2 is out now. Would be great if you could give it a spin and let me know if there's any more issues.

**Comment by hazzik** [comment]
@MichaelKetting thanks. I've tested it, all looks good: https://github.com/nhibernate/nhibernate-core/pull/568/commits/b6c2e2851b823bc9cc1518b222536bcbce177992

Also, RMLNQFETCH-21 shall be closed now

**Comment by MichaelKetting** [comment]
@hazzik Great, thanks. RTM version is out now.

**Pull request merged**